model/reliability/query_test.go

```diff
@@ -252,6 +291,144 @@ func TestGetTaskStatsTwoDocuments(t *testing.T) {
 	assert.Equal(docs[1].SuccessRate, float64(.42))
 }
 
+func GetTaskReliability(t *testing.T) {
+	requesters := []string{
+		evergreen.PatchVersionRequester,
+		evergreen.GithubPRRequester,
+		evergreen.MergeTestRequester,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	withCancelledContext := func(ctx context.Context, fn func(context.Context)) {
+		ctx, cancel := context.WithCancel(ctx)
+		cancel()
+		fn(ctx)
+	}
+
+	task1 := "task1"
+	task2 := "task2"
+
+	variant1 := "v1"
+	variant2 := "v2"
+
+	distro1 := "d1"
+	distro2 := "d2"
+
+	for opName, opTests := range map[string]func(ctx context.Context, t *testing.T, env evergreen.Environment){
+		"GetTaskReliability": func(ctx context.Context, t *testing.T, env evergreen.Environment) {
+			for testName, testCase := range map[string]func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter){
+				"No Matches": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.NotEqual(len(docs), 0)
+					})
+				},
+				"Invalid Combination task1": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task1}
+						filter.StatsFilter.BuildVariants = []string{variant2}
+						filter.StatsFilter.Distros = []string{distro2}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 0)
+					})
+				},
+				"Invalid Combination task2": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task2}
+						filter.StatsFilter.BuildVariants = []string{variant1}
+						filter.StatsFilter.Distros = []string{distro1}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 0)
+					})
+				},
+				"task1": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task1}
+						filter.StatsFilter.BuildVariants = []string{variant1, variant2}
+						filter.StatsFilter.Distros = []string{distro1, distro2}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 2)
+						for _, doc := range docs {
+							require.Equal(doc.TaskName, task1)
+							require.NotEqual(doc.BuildVariant, variant2)
+							require.NotEqual(doc.Distro, distro2)
+						}
+					})
+				},
+				"task2": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
```

Comment:
Isn't testing task1 then task2 testing the same thing? (that `GetTaskReliabilityScores` returns a score matching a filter containing Tasks, BuildVariants, and Distros)

Reply:
Yes, I'm just being a bit paranoid that the task matching portion is working.

Reply:
Ok